checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -1565,22 +1570,37 @@ private List<? extends VariableElement> getParametersOfInvocation(Node node) {
   }
 
   /**
-   * Does the method being invoked have a not-owning return type?
+   * Is the return type of the invoked method one that should be tracked?
    *
    * @param node a method invocation
-   * @return true iff the checker is not in no-lightweight-ownership mode and (1) the method has a
-   *     void return type, or (2) a NotOwning annotation is present on the method declaration
+   * @return true iff the checker is in no-lightweight-ownership mode, or the method has a
+   *     {@code @MustCallAlias} annotation, or (1) the method has a return type that needs to be
+   *     tracked (i.e., it has a non-empty {@code @MustCall} obligation and (2) the method
+   *     declaration does not have a {@code @NotOwning} annotation
    */
-  private boolean hasNotOwningReturnType(MethodInvocationNode node) {
+  private boolean shouldTrackReturnType(MethodInvocationNode node) {
     if (checker.hasOption(MustCallChecker.NO_LIGHTWEIGHT_OWNERSHIP)) {
       // Default to always transferring at return if not using LO, just like Eclipse does.
-      return false;
+      return true;
     }
     MethodInvocationTree methodInvocationTree = node.getTree();
     ExecutableElement executableElement = TreeUtils.elementFromUse(methodInvocationTree);
-    // void methods are "not owning" by construction
-    return (ElementUtils.getType(executableElement).getKind() == TypeKind.VOID)
-        || (typeFactory.getDeclAnnotation(executableElement, NotOwning.class) != null);
+    if (typeFactory.hasMustCallAlias(executableElement)) {
+      // assume tracking is required
+      return true;
+    }
+    TypeMirror type = ElementUtils.getType(executableElement);
+    // void or primitive-returning methods are "not owning" by construction
+    if (type.getKind() == TypeKind.VOID || type.getKind().isPrimitive()) {
+      return false;
+    }
+    TypeElement typeElt = TypesUtils.getTypeElement(type);
+    // no need to track if type has no possible @MustCall obligation
+    if (typeElt != null && typeFactory.getMustCallValue(typeElt).isEmpty()) {
+      return false;
```

Comment:
This is unsound, because it doesn't take into account potential `@MustCall` annotations on types that don't have default `@MustCall` annotations on their declarations. I wrote a failing test here: https://github.com/msridhar/checker-framework/pull/74

Reply:
Good catch!  Fixed in f77cda6de94b88ebed79a452f7d4962fe734734f, and I also added a test case for constructors.